File: Piece of cake/976. Largest Perimeter Triangle/main.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
   public:
    int largestPerimeter(vector<int>& A) {
        int n = A.size();
        sort(A.begin(), A.end());
        for (int i = n - 3; i >= 0; --i) {
            if (A[i] + A[i + 1] > A[i + 2]) return A[i] + A[i + 1] + A[i + 2];
        }
        return 0;
    }
};

int main() { return 0; }

File: Piece of cake/976. Largest Perimeter Triangle/main.rs
#![allow(dead_code)]
struct Solution;
impl Solution {
    pub fn largest_perimeter(mut a: Vec<i32>) -> i32 {
        a.sort_unstable();
        a.windows(3)
            .rev()
            .find(|&t| t[0] + t[1] > t[2])
            .map_or(0, |t| t[0] + t[1] + t[2])
    }
}

fn main() {}
